src/audio/component.c

```diff
@@ -193,6 +195,26 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 			ret = -EINVAL;
 		}
 		break;
+	case COMP_TRIGGER_SUSPEND:
+		if (dev->state == COMP_STATE_ACTIVE) {
+			dev->state = COMP_STATE_SUSPEND;
+		} else {
+			trace_comp_error("comp_set_state() error: "
+					 "wrong state = %u, "
+					 "COMP_TRIGGER_SUSPEND", dev->state);
+			ret = -EINVAL;
+		}
+		break;
+	case COMP_TRIGGER_RESUME:
+		if (dev->state == COMP_STATE_SUSPEND) {
+			dev->state = COMP_STATE_ACTIVE;
+		} else {
+			trace_comp_error("comp_set_state() error: "
+					 "wrong state = %u, "
+					 "COMP_TRIGGER_RESUME", dev->state);
+			ret = -EINVAL;
+		}
+		break;
```

Comment:
can you clarify who sends this TRIGGER_SUSPEND and TRIGGER_RESUME?
suspend/resume are loaded terms for anyone with a Linux/ALSA background, and I wonder if what you really wanted is an 'inactive' state.

Reply:
Initially I've added state attribute to the buffer, but such approach was too generic and we want to limit only the copy function. Copy is done only for components with state ACTIVE, so it became clear, that we should operate based on that. As for naming, I have no problem with changing it to INACTIVE, but it seemed reasonable to reuse already known state. The pipeline is in fact set to ACTIVE from the kernel perspective, so SUSPEND sounded like a reasonable choice, but as I understand it can be misunderstood as some kind of power related operation.
KPB component should trigger such state transition as it's the only one with the full knowledge of current state.
Nevertheless I can add new INACTIVE state and change this commit if that's your suggestion.

Reply:
@tlauda I like the proposal of using the component state instead of buffer states too as it can be used if we ever optimize the current topology to get rid of the intermediate buffers at the output of KPB. But I agree with Pierre that we should add a new state instead of using SUSPEND

Reply:
@plbossart @ranj063 Done.